Network/Network.cpp

```diff
@@ -625,154 +654,277 @@ namespace WPEFramework
-                    result = true;
-                }
-                else
-                    response["supported"] = iarmData.isSupported;
             }
+            else
+            {
+                response["supported"] = iarmData.isSupported;
+                result = true;
+                returnResponse(result)
+            }
+            if (IARM_RESULT_SUCCESS == IARM_Bus_Call(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETSRVMGR_API_setIPSettings, (void *) &iarmData, sizeof(iarmData)))
+            {
+                response["supported"] = iarmData.isSupported;
+                result = true;
+            }
+            else
+                response["supported"] = iarmData.isSupported;
 
             returnResponse(result)
         }
 
-        uint32_t Network::getIPSettings(const JsonObject& parameters, JsonObject& response)
+
+	uint32_t Network::getIPSettings(const JsonObject& parameters, JsonObject& response)
         {
+            JsonObject internal;
+            JsonObject InternalResponse;
             bool result = false;
-            string interface = "";
+            string interface;
+            string ipversion;
+            internal ["interface"] = interface;
+            internal ["ipversion"] = ipversion;
+
+            if (getIPSettingsInternal(internal,InternalResponse))
+            {
+                response["interface"] = InternalResponse["interface"];
+                response["ipversion"] = InternalResponse["ipversion"];
+                response["autoconfig"] = InternalResponse["autoconfig"];
+                response["ipaddr"] = InternalResponse["ipaddr"];
+                response["netmask"] = InternalResponse["netmask"];
+                response["gateway"] = InternalResponse["gateway"];
+                response["primarydns"] = InternalResponse["primarydns"];
+		response["secondarydns"] = InternalResponse["secondarydns"];
+                result = true;
+            }
+            returnResponse(result)
+        }
+
+	uint32_t Network::getIPSettings2(const JsonObject& parameters, JsonObject& response)
+        {
+            JsonObject internal;
+	    JsonObject InternalResponse;
+            NetworkManager_GetIPSettings_ErrorCode_t errCode = NETWORK_IPADDRESS_ACQUIRED;
+	    bool result = false;
+            string interface;
+            string ipversion;
+            internal ["interface"] = interface;
+            internal ["ipversion"] = ipversion;
+			
+	    if (getIPSettingsInternal(internal,InternalResponse))
+            {
+                /*If the device was configured to use autoconfig IP but device does not have valid IP yet Could be the Router does not have DHCP Server running or
+                 * the device is in the process of acquiring it from the router),it must return only the autoconfig and the interface name. */
+		 if ( InternalResponse["ipaddr"].Content() == Core::JSON::Variant::type::EMPTY || errCode == NETWORK_IPADDRESS_NOTFOUND )
```

Comment:
From where `errCode` is updated? the internal function does not seem to update this.